frame/contracts/src/tests.rs

```diff
@@ -1759,82 +1779,29 @@ fn lazy_removal_partial_remove_works() {
 }
 
 #[test]
-fn lazy_removal_does_no_run_on_full_block() {
-	let (code, hash) = compile_module::<Test>("self_destruct").unwrap();
+fn lazy_removal_does_no_run_on_full_queue_and_full_block() {
```

Comment:
How about also adding the test case when the deletion queue is not full but the block is full (so that `on_idle` runs but doesn't clear the queue)?

Reply:
> How about also adding the test case when the deletion queue is not full but the block is full? (which probably has two sub-cases: 1) `on_idle` just doesn't run at all, and 2) `on_idle` runs but `remaining_weight` is so low that it doesn't remove anything from deletion queue)

Thanks for the hints! Actually the case 2) that `remaining_weight` is to low for deletion is certainly easy to implement! I'm not too sure about case 1), it should be a general feature of the `on_idle` hook on Substrate, that it doesn't get executed when `remaining_weight` is zero. So I'd assume that this case is covered by tests somewhere else in Substrate?

Reply:
> So I'd assume that this case is covered by tests somewhere else in Substrate?

Could be non zero but still too low to remove a single item.

Reply:
Yes, but that what I said. I can implement case 2) but not case 1).

I now added the test `lazy_removal_does_no_run_on_low_remaining_weight` which runs `on_idle` with a remaining weight which is too low, then `on_initialize` with a deletion queue which is not full and finally `on_idle` again with a remaining weight which is high enough for deletion.